### PR TITLE
Bubble sorting algorithm

### PR DESCRIPTION
**Pull request by El-Potato-Slayer**
## Advanced Building Blocks - Bubble Sort

In this milestone, we:
- Added a bubble sorting algorithm that works as a normal method
- Added a bubble sorting algorithm that works a block and makes use of the "yield" keyword

### CODE REVIEW

**Comment by bertil291utn** [comment]
# **Status:** Invalid 🚫

## Hey guys I hope you're having a great time

I see all your effort work on this project however I have to invalid due to: 


### Summary

- Edit your pull request message, Follow this [guideline](https://cdn.filestackcontent.com/osvFJEkQHWQsgDe0PJoQ)
- Please leave just one screenshot in README file

**Note:** to achieve all this check the link that is above this 
<hr>

Guys work in the requested changes above and request another review.
Good luck. 

> “Never stop, never conform until what is good becomes even better, and what’s better becomes excellent.”

**Comment by bertil291utn** [comment]
## HEY guys
- Sorry about the above review, was a mistake to mark as invalid, however, while you're in there work on those 2 things and request another review.

I'm sorry for the inconveniences 🙏

File: bubble_sorting.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      break if index + 1 == arr.length
+
+      arr[index] > arr[index.next] && (arr[index], arr[index.next] = arr[index.next], arr[index])
+    end
+    arr
+  end
+end
+
+puts bubble_sort([3, 0, 128 - 3, -65, 45, 4, 5, 1, 10, 2])
+
```
Comment (bertil291utn) [inline bubble_sorting.rb]: ## Debugging code 
- No debugging instructions in committed code (as discussed [here](https://questions.microverse.org/t/debugging-instructions-code/346)) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/section/debugging_requirement.png) from Pathwright)
  - [ ] Please remove a debugging code

File: bubble_sorting.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  arr.each_with_index do |_value, _i|
+    arr.each_with_index do |_val, index|
+      break if index.next == arr.length
+
+      difference = yield(arr[index], arr[index.next])
+      difference.positive? && (arr[index], arr[index.next] = arr[index.next], arr[index])
+    end
+  end
+  puts arr
+end
```
Comment (bertil291utn) [inline bubble_sorting.rb]: ## Debugging code 
- No debugging instructions in committed code (as discussed [here](https://questions.microverse.org/t/debugging-instructions-code/346)) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/section/debugging_requirement.png) from Pathwright)
  - [ ] Please remove a debugging code `p`, and return only the arr var

**Review by bertil291utn — CHANGES_REQUESTED** [review]
# **Status:** Changes requested 🟠 
<p align="center"> image <br><strong>Not yet</strong> </p>

# Hey guys, you've been doing a great job, and also I hope you enjoyed coding.

### Your work is amazing you master it GitHub Flow you have a professional README file. 

However, in order to follow all the guidelines, you required to do these changes, listed below:

- [ ] Remove all debugging code. Review inline comments below this comment box

### Summary

-  Don't forget to work with inline comments below this comment box.

***

Guys work in the requested changes above and request another review.
Good luck. 

> “Never stop, never conform until what is good becomes even better, and what’s better becomes excellent.”

**Review by christianotieno — APPROVED** [review]
# PROJECT APPROVED 🍾 
- All the required changes have been met in this project. I will go ahead and mark this as complete.

Happy Coding Team! 👍

**Pull request merged**